tests/unit/test_app.py

```diff
@@ -773,3 +781,192 @@ def test_eq_non_default_configurations(self):
+            'Statement': []
+        }
+    }
+
+    @auth_app.authorizer()
+    def builtin_auth(auth_request):
+        return response
+
+    actual = builtin_auth(event, None)
+    assert actual == response
+
+
+def test_can_specify_extra_auth_attributes():
+    auth_app = app.Chalice('builtin-auth')
+
+    @auth_app.authorizer(ttl_seconds=10, execution_role='arn:my-role')
+    def builtin_auth(auth_request):
+        pass
+
+    handler = auth_app.builtin_auth_handlers[0]
+    assert handler.ttl_seconds == 10
+    assert handler.execution_role == 'arn:my-role'
+
+
+def test_can_return_auth_response():
+    event = {
+        'type': 'TOKEN',
+        'authorizationToken': 'authtoken',
+        'methodArn': 'arn:aws:execute-api:us-west-2:1:id/dev/GET/a',
+    }
+    auth_app = app.Chalice('builtin-auth')
+
+    response = {
+        'context': {},
+        'principalId': 'principal',
+        'policyDocument': {
+            'Version': '2012-10-17',
+            'Statement': [
+                {'Action': 'execute-api:Invoke',
+                 'Effect': 'Allow',
+                 'Resource': [
+                     'arn:aws:execute-api:us-west-2:1:id/dev/%s/a' %
+                     method for method in app.AuthResponse.ALL_HTTP_METHODS
+                 ]}
+            ]
+        }
+    }
+
+    @auth_app.authorizer()
+    def builtin_auth(auth_request):
+        return app.AuthResponse(['/a'], 'principal')
+
+    actual = builtin_auth(event, None)
+    assert actual == response
+
+
+def test_auth_response_serialization():
+    method_arn = (
+        "arn:aws:execute-api:us-west-2:123:rest-api-id/dev/GET/needs/auth")
+    request = app.AuthRequest('TOKEN', 'authtoken', method_arn)
+    response = app.AuthResponse(routes=['/needs/auth'], principal_id='foo')
+    response_dict = response.to_dict(request)
+    expected = [
+        method_arn.replace('GET', method)
+        for method in app.AuthResponse.ALL_HTTP_METHODS
+    ]
+    assert response_dict == {
+        'policyDocument': {
+            'Version': '2012-10-17',
+            'Statement': [
+                {
+                    'Action': 'execute-api:Invoke',
+                    'Resource': expected,
+                    'Effect': 'Allow'
+                }
+            ]
+        },
+        'context': {},
+        'principalId': 'foo',
+    }
+
+
+def test_auth_response_can_include_context(auth_request):
+    response = app.AuthResponse(['/foo'], 'principal', {'foo': 'bar'})
+    serialized = response.to_dict(auth_request)
+    assert serialized['context'] == {'foo': 'bar'}
+
+
+def test_can_use_auth_routes_instead_of_strings(auth_request):
+    expected = [
+        "arn:aws:execute-api:us-west-2:123:rest-api-id/dev/GET/a",
+        "arn:aws:execute-api:us-west-2:123:rest-api-id/dev/GET/a/b",
+        "arn:aws:execute-api:us-west-2:123:rest-api-id/dev/POST/a/b",
+    ]
+    response = app.AuthResponse(
+        [app.AuthRoute('/a', ['GET']),
```

Comment:
Do you think it is worth adding a test that mixes string and AuthRoute in the AuthResponse list?